### ISSUE TITLE
Should a nested interactive element with tabindex="-1" pass?

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#2954]
@melaniephilipp  pointed me at this beauty yesterday:

```html
<div role="button" tabindex="0" aria-expanded="false">
  <a href="#" tabindex="-1"> Login </a>
</button>
```

To my surprise, if I put the focus on the link, Safari / VoiceOver will announce the button, including aria-expanded. I know focus was on the link. I can see that through `document.activeElement`. Safari+VO was specifically the case for which we started failing this. Other browsers don't treat children of buttons as presentational, so I suspect this is going to work fine there too.... although they'll announce the it as a link, without including info on aria-expanded.

We'll need to test this in other AT, and I'd like to experiment a little more with that. Maybe figure out what happens if there is more than one link in the button, or if there is some text outside the link, still in the button.

I suspect that if VO encounters an element that it can't find in the accessibility tree it might look for a parent that is included instead. I would like to know why this works, so we can make a more informed decision on this.

**Comment by scottaohara** [comment]
FWIW, I came across a similar example in the wild just this week which definitely poses some issues even without a screen reader / other AT being used.

https://codepen.io/scottohara/pen/GREyEvo

Here the author has an ARIA button to expand/collapse a section of content (not included in the test case) that wraps the link.  The link has a valid href.  (target=_blank added to reduced test case to help showcase behavior).

The link of course cannot be accessed because of the negative tabindex with keyboard, but it can be activated on mouse click.  Keyboard users can expand/collapse the associated content.

Mouse users can activate the link (and due to the way this was coded, the clicking will also expand content... but that doesn't actually matter since the browser has navigated user to the new page before the expanded content can even be reviewed).

**Comment by WilcoFiers** [comment]
After dozens of conversations, the conclusion on this is that the only exception we're going to allow things like this:

```html
<button><span tabindex="-1">Hello world</span></button>
```

Any other focusable content inside interactive elements (except links) will continue to fail axe-core.

**Comment by padmavemulapati** [comment]
Verified with the latest develop branch code, when role=text and tabindex="-1",
its not failing
![image](https://user-images.githubusercontent.com/47409579/147926192-6a64b897-5b16-49eb-a736-2f9e1990c617.png)
